rosetta/services/transaction_parser.go

```diff
@@ -52,10 +58,24 @@ func (tp *transactionsParser) parseTx(eTx *data.FullTransaction, isInPool bool)
 }
 
 func (tp *transactionsParser) createRosettaTxFromUnsignedTx(eTx *data.FullTransaction) (*types.Transaction, bool) {
+	// TODO check if we have a SCR that call another contract
 	if eTx.Value == "0" {
 		return nil, false
 	}
 
+	switch {
+	case eTx.GasLimit != 0:
```

Comment:
I think this is not sufficient to identify the gas refund SCRs
I think also for SCR from SC to SC will have gasLimit.
Can you please confirm with @sasurobert and find a way to identify specifically the SCRs only with gas refund.

Reply:
Async calls indeed have a gas limit as well.

I've added this task - to improve the robustness of handling SCRs:
https://elrondnetwork.atlassian.net/browse/EN-7896

@sasurobert, can we perhaps add a discriminant on SCRs - such as `type = refund`, `type = SC call`, `type = transferValue`?

Reply:
adding any new type to SCR / transaction will be breaking changes - thus need backward compatibility checks everywhere. We have enough information to decipher which is the gasRefund SCR. It must have sender smart contract, receiver the original sender, scr.Nonce = tx.Nonce + 1. it is the only one which has hex.encode(vmOutput.returnCode).
https://github.com/ElrondNetwork/elrond-go/blob/3d03e6e38474a787bc9b91b18ddff16708488010/process/smartContract/process.go#L1326 check code here.